crates/aranya-daemon/src/sync.rs

```diff
@@ -78,6 +96,40 @@ impl SyncPeers {
             error!(?e, "error removing peer from syncer");
             return Err(e);
         }
+
+        self.cfgs.remove(&(addr, graph_id));
+
+        Ok(())
+    }
+
+    /// Sync with a peer immediately.
+    pub async fn sync_now(
+        &self,
+        addr: Addr,
+        graph_id: GraphId,
+        maybe_cfg: Option<SyncPeerConfig>,
+    ) -> Result<()> {
+        let _cfg = match maybe_cfg {
+            Some(c) => c,
+            None => self
+                .cfgs
+                .get(&(addr, graph_id))
+                .cloned()
+                .unwrap_or_default(),
+        };
```

Comment:
Or use a default then set to something else if the config is set.
```suggestion
let mut cfg = SyncPeerCfg::default()
if let Some(sync_cfg) = self.cfgs.get(...)... {
    cfg = sync_cfg
}
```

Reply:
Generally, it's better to generate values until you need them. With this suggestion, we'd generate the default config and may not even use it.

Reply:
🤓 
```suggestion
        let _cfg = maybe_cfg
            .or_else(|| self.cfgs.get(&(addr, graph_id)).cloned())
            .unwrap_or_default();
```

Reply:
Nice